tests/cupy_tests/math_tests/test_sumprod.py

```diff
@@ -118,6 +118,16 @@ def test_sum_axes4(self, xp, dtype):
         a = testing.shaped_arange((20, 30, 40, 50), xp, dtype)
         return a.sum(axis=(0, 2, 3))
 
+    @testing.for_all_dtypes_combination(names=['src_dtype', 'dst_dtype'])
+    @testing.numpy_cupy_allclose()
+    def test_sum_dtype(self, xp, src_dtype, dst_dtype):
+        if not xp.can_cast(src_dtype, dst_dtype) \
+                or [xp.dtype(src_dtype).kind,
+                    xp.dtype(dst_dtype).kind].count('c') == 1:
```

Comment:
I guess it should be `src_dtype == xp.float16 and xp.dtype(dst_dtype).kind == 'c'`?